Assets/Scripts/Interactables/CocktailMiniGame.cs

```diff
@@ -0,0 +1,374 @@
+using FMS.TapperRedone.Characters;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI; // For UI elements
+
+using FMS.TapperRedone.Data;
+using FMS.TapperRedone.Inventory;
+using FMS.TapperRedone.Managers;
+
+namespace FMS.TapperRedone.Interactables
+{
+
+    public class CocktailMiniGame : InteractableObject
+    {
+
+        // References to the sprites for the arrows
+        public Sprite leftUIPressed;
+        public Sprite leftUIUnpressed;
+        public Sprite rightUIPressed;
+        public Sprite rightUIUnpressed;
+
+        public SpriteRenderer cocktailShakerRenderer; // Assign this with the Cocktail Shaker's Sprite Renderer in the Inspector
+        public GameObject leftArrow; // Assign the LeftArrow object here
+        public GameObject rightArrow; // Assign the RightArrow object here
+
+        private PlayerInteraction player; // Store player reference
```

Comment:
Try to separate into sections things that are public/serializable (aka modifiable in editor) from private (aka only modifiable within this class's code)